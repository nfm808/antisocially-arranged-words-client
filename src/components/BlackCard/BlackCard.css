.BlackCard {
  width: 120px;
  height: 180px;
  background: black;
  border-radius: 5px;
  box-shadow: 0px 10px 30px -5px rgba(0, 0, 0, 0.3);
  transition: box-shadow 0.5s;
  will-change: transform;
  color: white;
  margin: 0 auto;
}

.BlackCard--text {
  padding: 5px;
  text-align: left;
}