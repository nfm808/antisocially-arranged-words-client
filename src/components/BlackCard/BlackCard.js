import React from 'react';
import PropTypes from 'prop-types';
import { useSpring, animated } from 'react-spring';
import './BlackCard.css';

const calc = (x, y) => [-(y - window.innerHeight / 2) / 100, (x - window.innerWidth / 2) / 100, 1.1]
const trans = (x, y, s) => `perspective(600px) rotateX(${x}deg) rotateY(${y}deg) scale(${s})`

export default function BlackCard(props) {
  const [prop, set] = useSpring(() => ({ xys: [0, 0, 1], config: { mass: 7, tension: 200, friction: 80 } }))
  return (
        <animated.div 
          className="BlackCard"
          onMouseMove={({ clientX: x, clientY: y }) => set({ xys: calc(x, y) })}
          onMouseLeave={() => set({ xys: [0, 0, 1] })}
          style={{ transform: prop.xys.interpolate(trans) }}>
          <h5 className="BlackCard--text">{props.card.text}</h5>
          <p className="BlackCard--text">{props.card.playerName && props.card.playerName}</p>
        </animated.div>
  )
}

BlackCard.propTypes = {
  card: PropTypes.object,
}
