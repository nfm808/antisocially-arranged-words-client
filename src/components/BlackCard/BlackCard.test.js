import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import BlackCard from './BlackCard';

describe('<BlackCard />', () => {
  const testCard = {
    id: 1,
    text:'test card text'
  }
  it('renders without crashing', () => {
    const wrapper = shallow(<BlackCard card={testCard}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
})