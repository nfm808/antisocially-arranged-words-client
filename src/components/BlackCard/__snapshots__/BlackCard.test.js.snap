// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlackCard /> renders without crashing 1`] = `
<ForwardRef
  className="BlackCard"
  onMouseLeave={[Function]}
  onMouseMove={[Function]}
  style={
    Object {
      "transform": AnimatedInterpolation {
        "attach": [Function],
        "calc": [Function],
        "children": Array [],
        "detach": [Function],
        "payload": Array [
          AnimatedValue {
            "animatedStyles": Set {},
            "children": Array [],
            "done": false,
            "lastPosition": 0,
            "lastTime": undefined,
            "lastVelocity": undefined,
            "payload": undefined,
            "setValue": [Function],
            "startPosition": 0,
            "startTime": 1563788444582,
            "value": 0,
          },
          AnimatedValue {
            "animatedStyles": Set {},
            "children": Array [],
            "done": false,
            "lastPosition": 0,
            "lastTime": undefined,
            "lastVelocity": undefined,
            "payload": undefined,
            "setValue": [Function],
            "startPosition": 0,
            "startTime": 1563788444582,
            "value": 0,
          },
          AnimatedValue {
            "animatedStyles": Set {},
            "children": Array [],
            "done": false,
            "lastPosition": 1,
            "lastTime": undefined,
            "lastVelocity": undefined,
            "payload": undefined,
            "setValue": [Function],
            "startPosition": 1,
            "startTime": 1563788444582,
            "value": 1,
          },
        ],
      },
    }
  }
>
  <h5
    className="BlackCard--text"
  >
    test card text
  </h5>
</ForwardRef>
`;
